File: 4.2.Singleton/Exercises43/app/src/main/java/net/braniumacademy/exercises43/MainActivity.java
package net.braniumacademy.exercises43;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import net.braniumacademy.exercises1.CountryActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button btnExercises1 = findViewById(R.id.btn_exercises1);
        btnExercises1.setOnClickListener(v -> {
            Intent intent = new Intent(this, CountryActivity.class);
            startActivity(intent);
        });
    }
}

File: 4.2.Singleton/SingletonPattern/app/src/main/java/net/braniumacademy/singletonpattern/MainActivity.kt
package net.braniumacademy.singletonpattern

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val viewModel = UserViewModel.getInstance()
        val textInfo: TextView = findViewById(R.id.text_user_info)

        val btnStart = findViewById<Button>(R.id.btn_start_activity)
        btnStart.setOnClickListener {
            Intent(this, OtherActivity::class.java).apply {
                startActivity(this)
            }
        }

        viewModel.user.observe(this) {
            val message = "Full name: ${it.fullName}"
            textInfo.text = message
        }

        Log.e("MainActivity ==> ", "${viewModel.hashCode()}")
    }
}